python/engine/PythonEngine.cpp

```diff
@@ -0,0 +1,170 @@
+#include "PythonEngine.hpp"
+#include <utilities/core/ApplicationPathHelpers.hpp>
+#include <fmt/format.h>
+#include <stdexcept>
+#include <string>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
+
+#include <Python.h>
+#include <SWIGPythonRuntime.hxx>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
+
+namespace openstudio {
+
+static constexpr auto importScript = R"(
+import sys
+sys.path.insert(0, '{}')
+import {}
+)";
+
+PythonEngine::PythonEngine(int argc, char *argv[])
+    : ScriptEngine(argc, argv), program(Py_DecodeLocale("python3.8", nullptr)) {
+  Py_SetProgramName(program); // optional but recommended
+
+  Py_Initialize();
+  importOpenStudio();
+}
+
+PythonEngine::~PythonEngine() {
+  if (Py_FinalizeEx() < 0) {
+    exit(120);
+  }
+  PyMem_RawFree(program);
+}
+
+void PythonEngine::importOpenStudio() {
+  if (moduleIsRunningFromBuildDirectory()) {
+    const auto bindingsDir = getOpenStudioModuleDirectory();
+    exec(fmt::format(importScript, bindingsDir.string(), "openstudiodev"));
```

Comment:
IS exec better than PyRun_SimpleString ?

Reply:
I think ScriptEngine::exec is good because it is a nice abstraction that is the same among scripting languages. Internally PyRun_SipleString would be ok.